portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,97 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.List; 
+import java.util.ArrayList;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+import java.util.HashMap;
+import java.util.Map;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private HashMap<String, HashMap<String,Object>> information;  
+  private List<String> messages;
+
+  @Override
+  public void init(){
+      messages = new ArrayList<>();
+      information = new HashMap<String, HashMap<String,Object>>();
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Thomas!</h1>");
+    
+    String json = convertToJsonUsingGson(information);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
+      
+      information = checkValidity(request);
+
+      if (((Boolean) information.get("nameInfo").get("error") == true) || 
```

Comment:
Is casting to boolean required here given you've set the values to valid boolean true/false values in the checkValidity function?

Reply:
Because I created my object as Map<String, Object>, where the values are objects, Java expects an Object. When I use the equal sign, there is an error of comparing an Object and Boolean value, which is why I had to use a Boolean cast. I'm not sure there is an alternative way, since I'm still new to Java syntax.

Reply:
done.